core/common/iterator/src/main/java/org/eclipse/rdf4j/common/iteration/CloseableIteration.java

```diff
@@ -23,9 +27,32 @@
  *   // process the exception that can be thrown while processing.
  * }
  * </pre>
+ *
+ * @implNote In the future this interface will stop extending {@link Iteration} and instead declare the same interface
+ *           methods directly. The interface will also stop requiring implementations to automatically close when
+ *           exhausted, instead making this an optional feature and requiring the user to always call close.
  */
 public interface CloseableIteration<E, X extends Exception> extends Iteration<E, X>, AutoCloseable {
 
+	/**
+	 * Convert the results to a Java 8 Stream.
+	 *
+	 * @return stream
+	 */
+	default Stream<E> stream() {
+		return StreamSupport
+				.stream(new CloseableIterationSpliterator<>(this), false)
+				.onClose(() -> {
+					try {
+						close();
+					} catch (RuntimeException e) {
+						throw e;
+					} catch (Exception e) {
+						throw new RuntimeException(e);
+					}
+				});
```

Comment:
I think you can replace this by using the new method you added to `Iterations`, instead.